### PR TITLE
Plaintext channel builder patch by BanzaiCloud

### PR DESCRIPTION
**Pull request by schultheiszn**
*Custom transport layer built to enable to achieve a transparent way of authorization in which the application uses [CertificationRequestInfo](https://tools.ietf.org/html/rfc2986#section-4) fields to identify clients*

*All unit test both in clients and core module and Integration tests as well are OK*

### CODE REVIEW

File: clients/src/main/java/org/apache/kafka/common/network/BanzaiChannelBuilder.java [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+            return new KafkaChannel(id, transportLayer, authenticatorCreator, maxReceiveSize,
+                    memoryPool != null ? memoryPool : MemoryPool.NONE);
+        } catch (Exception e) {
+            log.warn("Failed to create channel due to ", e);
+            throw new KafkaException(e);
+        }
+    }
+
+    /**
+     *
```
Comment (hi-im-aren) [inline clients/src/main/java/org/apache/kafka/common/network/BanzaiChannelBuilder.java]: Missing comments

File: clients/src/test/resources/log4j.properties [hunk 2]
```diff
@@ -12,7 +12,7 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-log4j.rootLogger=OFF, stdout
+log4j.rootLogger=DEBUG, stdout
```
Comment (hi-im-aren) [inline clients/src/test/resources/log4j.properties]: Is this necessary? shouldn't Information level be enough?

File: clients/src/main/java/org/apache/kafka/common/network/BanzaiTransportLayer.java [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+    public long read(ByteBuffer[] dsts) throws IOException {
+        return read(dsts, 0, dsts.length);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public int read(ByteBuffer dst) throws IOException {
+        if (log.isTraceEnabled())
```
Comment (hi-im-aren) [inline clients/src/main/java/org/apache/kafka/common/network/BanzaiTransportLayer.java]: There are multiple instances of checking this condition in the code, I'm not sure it's needed

File: clients/src/main/java/org/apache/kafka/common/network/BanzaiTransportLayer.java [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+        int i = offset;
+        while (i < length) {
+            if (dsts[i].hasRemaining()) {
+                int read = read(dsts[i]);
+                if (read > 0)
+                    totalRead += read;
+                else
+                    break;
+            }
+            if (!dsts[i].hasRemaining()) {
```
Comment (hi-im-aren) [inline clients/src/main/java/org/apache/kafka/common/network/BanzaiTransportLayer.java]: You can just use an else branch here, so the condition does not get evaluated twice each loop

File: clients/src/main/java/org/apache/kafka/common/network/BanzaiTransportLayer.java (hunk 3 above)
Comment (stoader) [reply clients/src/main/java/org/apache/kafka/common/network/BanzaiTransportLayer.java]: if `log.trace` checks whether trace log level is enabled than `if (log.isTraceEnabled())` is redundant here.

**Pull request closed without merging**